src/screens/explore-screen/components/CollectionCard.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { useCallback } from 'react'
+
+import { UserCollection } from 'audius-client/src/common/models/Collection'
+import { albumPage, playlistPage } from 'audius-client/src/utils/route'
+import { push } from 'connected-react-router'
+import { StyleProp, ViewStyle } from 'react-native'
+
+import { Card } from 'app/components/card'
+import { useDispatchWeb } from 'app/hooks/useDispatchWeb'
+import { formatCount } from 'app/utils/format'
+
+const formatPlaylistCardSecondaryText = (saves: number, tracks: number) => {
+  const savesText = saves === 1 ? 'Favorite' : 'Favorites'
+  const tracksText = tracks === 1 ? 'Track' : 'Tracks'
+  return `${formatCount(saves)} ${savesText} • ${tracks} ${tracksText}`
+}
+
+type CollectionCardProps = {
+  collection: UserCollection
+  style: StyleProp<ViewStyle>
+}
+
+export const CollectionCard = ({ collection, style }: CollectionCardProps) => {
+  const dispatchWeb = useDispatchWeb()
+  const goToRoute = useCallback((route: string) => dispatchWeb(push(route)), [
+    dispatchWeb
+  ])
+
+  const collectionPage = collection.is_album ? albumPage : playlistPage
+
+  return (
+    <Card
+      style={style}
+      id={collection.playlist_id}
+      type={'collection'}
+      imageSize={collection._cover_art_sizes}
+      primaryText={collection.playlist_name}
+      secondaryText={formatPlaylistCardSecondaryText(
+        collection.save_count,
+        collection.playlist_contents.track_ids.length
+      )}
+      onPress={() =>
+        goToRoute(
```

Comment:
same thing here! also, maybe we make a helper hook like `useRoute` that does all the goToRoute logic?

Reply:
Good call. Looks like we have `usePushRouteWeb` hook so I'll use that here